PasswordVerifier/src/main.java

```diff
@@ -0,0 +1,134 @@
+        
+        ArrayList<Boolean> testResults = new ArrayList<Boolean>();
+        for(int i = 0; i < testCases.size(); ++i) {
+            testResults.add(utils.isValid(testCases.get(i).getKey()));
+        }
+        ArrayList<Integer> errorIndicies = new ArrayList<Integer>();
+        for(int i = 0; i < testResults.size(); ++i) {
+            if(testResults.get(i) != testCases.get(i).getValue().booleanValue())
+                errorIndicies.add(new Integer(i));
+        }
+        if(errorIndicies.size() == 0) {
+            System.out.println("no errors. Good job!");
+            return true;
+        }
+        else {
+            System.out.println("errors at indicies:");
+            System.out.println(errorIndicies.toString());
+            return false;
+        }
+    }
+
+    /*
+     * stress tests isValid function. prints time to check test cases
+     * @param count number of test cases to run
+     */
+    public static void stressTest(int count, boolean logging) {
+        final int minLength = 5;
+        final int maxLength = 20;
+        int nextLength;
+        char nextChar;
+        char[] nextString;
+        Random myRandom = new Random();
+        String[] testCases = new String[count];
+        boolean[] testResults = new boolean[count];
+        for (int i = 0; i < testCases.length; ++i) {
+            nextLength = (myRandom.nextInt(maxLength - minLength + 1) + minLength);
+            nextString = new char[nextLength];
+            for (int j = 0; j < nextString.length; ++j) {
+                //arbitrary ascii range containing mostly valid characters
+                nextChar = (char) (myRandom.nextInt(100) + 27);
+                nextString[j] = nextChar;
+            }
+            testCases[i] = new String(nextString);
+        }
+        
+        long startTime = System.nanoTime();
+        for (int i = 0; i < testCases.length; ++i) {
+            testResults[i] = utils.isValid(testCases[i]);
+        }
+        long endTime = System.nanoTime();
+        long duration = (endTime - startTime);
+        System.out.println("time to check: " + duration + " nanoseconds");
+        
+        if(logging) {
+	        PrintWriter myWriter = null;
+	        try {
+	        	myWriter = new PrintWriter("log.txt", "UTF-8");
+	        	for (int i = 0; i < testCases.length; ++i) {
+	        		myWriter.println(testCases[i] + ": " + testResults[i]);
+	        	}
+	        } catch (FileNotFoundException e) {
+	        	e.printStackTrace();
+	        } catch (UnsupportedEncodingException e) {
+	        	e.printStackTrace();
+	        } finally {
+	        	myWriter.close();
+	        }
+        }
+    }
+    
+    public static void main(String[] args) {
+    	Properties myProperties = new Properties();
+    	FileInputStream input = null;
+    	boolean logging = false;
+    	try {
+    		input = new FileInputStream("/Users/benjaminkuschner/Documents/SecurityProjects/PasswordVerifier/src/config.properties");
```

Comment:
This path specification could be in the config file so that if a user wants to change where it's stored they don't have to recompile

Reply:
I am not sure if this is possible, since how would my program retrieve the file path of the config file, if that file path is only in the config file?